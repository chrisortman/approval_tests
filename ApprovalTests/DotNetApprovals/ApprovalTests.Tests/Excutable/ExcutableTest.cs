using System;
using System.Collections.Generic;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace ApprovalTests.Tests.Excutable
{
	[TestFixture]
	[UseReporter(typeof(QuietReporter))]
	public class ExcutableTest
	{
		[Test]
		public void TestExecutableFailure()
		{
			Approvals.VerifyAll(RunExecutableApproval(), "Increased feedback on");
		}

		[Test]
		public void TestExecutableFailureWithPreviousApproval()
		{
			Approvals.VerifyAll(RunExecutableApproval(), "Increased feedback on");
		}
		[Test]
		public void TestExecutableSuccess()
		{
			Approvals.VerifyAll(RunExecutableApproval(), "Increased feedback on");
		}

		private List<string> RunExecutableApproval()
		{
			var output = new List<string>();

			try
			{
				NamerFactory.AdditionalInformation = "Inner";
				Approvals.VerifyWithCallback("Sam", s => output.Add(s));

			}
			catch (Exception)
			{

			}
			return output;
		}
	}
}