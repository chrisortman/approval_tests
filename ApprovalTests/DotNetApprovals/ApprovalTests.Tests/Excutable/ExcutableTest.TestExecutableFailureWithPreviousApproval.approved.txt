Increased feedback on[0] = I am
Increased feedback on[1] = Sam
